0.preprocess-sites/7.visualize-cell-summary.py

```diff
@@ -0,0 +1,249 @@
+#cell_count_df.to_csv(output_file, sep='\t', index=False)
+
+print(cell_count_df.shape)
+cell_count_df.head(10)
+
+# Graph: Cell count with all wells in same graph
+cell_count_gg = (
+    gg.ggplot(cell_count_df, gg.aes(x="Site", y="cell_count"))
+    + gg.geom_bar(gg.aes(fill="Cell_Quality"), stat="identity")
+    + gg.theme_bw()
+    + gg.theme(axis_text_x = gg.element_text(rotation=90, size=5))
+    + gg.xlab("Sites")
+    + gg.ylab("Cell Count")
+    + gg.scale_fill_manual(
+        name="Cell Quality",
+        labels = cell_category_list,
+        values = cell_category_colors
+    )
+)
+
+output_file = pathlib.Path(output_figuresdir, "all_cellpainting_cellquality_across_sites.png")
+#cell_count_gg.save(output_file, dpi=300, width=10, height=7, verbose=False)
+
+cell_count_gg
+
+#Same graph as above, separated by Well.
+well_order = cell_count_df['site'].unique()
+
+cell_count_df.loc[:, 'Well'] = (
+    pd.Categorical(
+        cell_count_df.Well,
+        categories=well_order
+    )
+)
+
+cell_count_gg_parsed = (
+    gg.ggplot(cell_count_df, gg.aes(x="Site", y="cell_count"))
+    + gg.geom_bar(gg.aes(fill="Cell_Quality"), stat="identity")
+    + gg.theme_bw()
+    + gg.theme(axis_text_x = gg.element_text(rotation=90, size=5),
+               strip_background = gg.element_rect(colour = "black", fill = "#fdfff4"))
+    + gg.xlab("Sites")
+    + gg.ylab("Cell Count")
+    + gg.scale_fill_manual(
+        name="Cell Quality",
+        labels = cell_category_order,
+        values = cell_category_colors
+      )
+    + gg.facet_wrap("~Well", nrow=2, ncol=2, drop=False, scales="free_x")
+    )
+
+output_file = pathlib.Path(output_figuresdir, "all_cellpainting_cellquality_across_sites_by_well.png")
+#cell_count_gg_parsed.save(output_file, dpi=300, width=10, height=7, verbose=False)
+
+cell_count_gg_parsed
+
+#  Total cells in each quality category
+all_count_df = pd.DataFrame(cell_count_df.groupby("Cell_Quality")["cell_count"].sum()).reset_index()
+all_count_df
+
+# Graph: Total cells in each quality category
+all_cells = all_count_df.cell_count.sum()
+
+total_cell_count_gg = (
+    gg.ggplot(all_count_df, gg.aes(x="Cell_Quality", y="cell_count"))
+    + gg.geom_bar(gg.aes(fill="Cell_Quality"), stat="identity")
+    + gg.theme_bw()
+    + gg.theme(axis_text_x = gg.element_text(rotation=90, size=9))
+    + gg.xlab("")
+    + gg.ylab("Cell Count")
+    + gg.ggtitle(f"{all_cells} Total Cells")
+    + gg.scale_fill_manual(
+        name="Cell Quality",
+        labels = cell_category_order,
+        values = cell_category_colors,
+    )
+)
+
+output_file = pathlib.Path(output_figuresdir, "total_cell_count.png")
+#total_cell_count_gg.save(output_file, dpi=300, width=5, height=6, verbose=False)
+
+total_cell_count_gg
+
+print(f"There are a total of {all_cells} cells in {batch}")
+
+# Total cell number by well
+all_well_count_df = (
+    pd.DataFrame(
+        cell_count_df
+        .groupby(
```

Comment:
Here's another groupby where I'm having an issue. cell_count is outputting as NaN